source/extensions/stat_sinks/common/statsd/statsd.h

```diff
@@ -107,7 +107,16 @@ class TcpStatsdSink : public Stats::Sink {
 public:
   TcpStatsdSink(const LocalInfo::LocalInfo& local_info, const std::string& cluster_name,
                 ThreadLocal::SlotAllocator& tls, Upstream::ClusterManager& cluster_manager,
-                Stats::Scope& scope, const std::string& prefix = getDefaultPrefix());
+                Stats::Scope& scope, absl::Status& creation_status,
+                const std::string& prefix = getDefaultPrefix());
+
+  /**
+   * This is a wrapper around the constructor to return StatusOr.
+   */
+  static absl::StatusOr<std::unique_ptr<TcpStatsdSink>>
+  create(const LocalInfo::LocalInfo& local_info, const std::string& cluster_name,
+         ThreadLocal::SlotAllocator& tls, Upstream::ClusterManager& cluster_manager,
+         Stats::Scope& scope, const std::string& prefix = getDefaultPrefix());
```

Comment:
this is following the style of https://github.com/envoyproxy/envoy/pull/37464

Reply:
i tried but turns out it's not feasible unfortunately - the constructor is accessed inside std::make_unique which blocks it from being private. One workaround for this is to use `std::unique_ptr<T>(new T)` like below but not sure if we want to use `new` just to make this private. wdyt?

```diff
diff --git a/source/extensions/stat_sinks/common/statsd/statsd.cc b/source/extensions/stat_sinks/common/statsd/statsd.cc
index 044d559c2d..c5423ef5be 100644
--- a/source/extensions/stat_sinks/common/statsd/statsd.cc
+++ b/source/extensions/stat_sinks/common/statsd/statsd.cc
@@ -210,8 +210,8 @@ TcpStatsdSink::create(const LocalInfo::LocalInfo& local_info, const std::string&
                       ThreadLocal::SlotAllocator& tls, Upstream::ClusterManager& cluster_manager,
                       Stats::Scope& scope, const std::string& prefix) {
   absl::Status creation_status;
-  auto sink = std::make_unique<TcpStatsdSink>(local_info, cluster_name, tls, cluster_manager, scope,
-                                              creation_status, prefix);
+  auto sink = std::unique_ptr<TcpStatsdSink>(new TcpStatsdSink(
+      local_info, cluster_name, tls, cluster_manager, scope, creation_status, prefix));
   RETURN_IF_ERROR(creation_status);
   return sink;
 }
```

Reply:
It looks like we use `new` in at least one other similar situation, in source/common/upstream/transport_socket_match_impl.cc:18.

Reply:
ok let me use new then - than you for taking a look at other places

Reply:
https://github.com/envoyproxy/envoy/pull/37491/commits/3c3e1e7ffe59154a75cb90bd7c8577cffd324a9c done

Reply:
thanks!